File: frontend/src/app/models/itinerary.ts
import { z } from "zod";
import * as L from 'leaflet';

export const ItinerarySchema = z.object({
  positions: z.array(z.custom<L.LatLngTuple>()).max(2),
  start_station_name: z.string().nullable().optional(),
  end_station_name: z.string().nullable().optional(),
  nb_acc: z.number().nullable().optional(),
  color: z.string().optional().nullable(),
  popup_text: z.string().optional().nullable(),
});

export type Itinerary = z.infer<typeof ItinerarySchema>;

File: backend/models/itinerary.py
#
#   Imports
#

from pydantic import BaseModel
from typing import Optional

#
#   Itinerary
#

class Itinerary(BaseModel):
    """
        One itinerary.
    """

    positions: list[list[float]]
    start_station_name: str
    end_station_name: str
    nb_acc: int
    popup_text : Optional[str] = None

